src/core/src/sys/interactions.rs

```diff
@@ -185,6 +185,20 @@ impl Interactions {
         }
     }
 
+    /// Get all pair interactions
+    ///
+    /// Using this function, you lose information
+    /// about which interaction belongs to which pair of particles.
+    pub fn all_pairs(&self) -> Vec<&PairInteraction> {
+        let res: Vec<&PairInteraction> = self.pairs
+            .values() // value is a Vec<&PairInteraction>
+            // flatten the vector -> return an iterator over
+            // Vec<&PairInteraction>
+            .flat_map(|i| i)
+            .collect();
+        res
```

Comment:
I believe this could be written

```rust
pub fn all_pairs(&self) -> Vec<&PairInteraction> {
    self.pairs.values() // value is a Vec<Vec<&PairInteraction>>
                        // flatten the vector -> return an iterator over
                        // Vec<&PairInteraction>
              .flat_map(|i| i)
              .collect()
}
```